op-acceptance-tests/tests/jovian/min_base_fee_test.go

```diff
@@ -0,0 +1,197 @@
+		ctx:                t.Ctx(),
+		log:                t.Logger(),
+		t:                  t,
+		require:            t.Require(),
+		l1Client:           l1EL,
+		l2Network:          l2Network,
+		l2EL:               l2EL,
+		systemConfig:       systemConfig,
+		originalMinBaseFee: originalMinBaseFee,
+	}
+}
+
+func (mbf *minBaseFee) checkCompatibility() bool {
+	_, err := contractio.Read(mbf.systemConfig.MinBaseFee(), mbf.ctx)
+	if err != nil {
+		mbf.t.Fail()
+		return false
+	}
+	return true
+}
+
+func (mbf *minBaseFee) getSystemOwner() *dsl.EOA {
+	priv := mbf.l2Network.Escape().Keys().Secret(devkeys.SystemConfigOwner.Key(mbf.l2Network.ChainID().ToBig()))
+	return dsl.NewKey(mbf.t, priv).User(mbf.l1Client)
+}
+
+func (mbf *minBaseFee) setMinBaseFee(minBaseFee uint64) {
+	owner := mbf.getSystemOwner()
+
+	_, err := contractio.Write(mbf.systemConfig.SetMinBaseFee(minBaseFee), mbf.ctx, owner.Plan())
+	mbf.require.NoError(err, "SetMinBaseFee transaction failed")
+
+	mbf.t.Logf("Set min base fee on L1: minBaseFee=%d", minBaseFee)
+}
+
+func (mbf *minBaseFee) verifyMinBaseFee(from *dsl.EOA, to *dsl.EOA, minBase *big.Int, shouldEnforce bool) {
+	// Simulate user transactions
+	for range 20 {
+		from.Transfer(to.Address(), eth.OneGWei)
+	}
+
+	info := mbf.getBlock()
+	prevBlockNum := info.NumberU64()
+	for range 5 {
+		n := mbf.getBlock()
+		mbf.require.True(n.NumberU64() > prevBlockNum, "block number should increase")
+
+		prevBlockNum = n.NumberU64()
+		feeCmp := n.BaseFee().Cmp(minBase)
+		if !shouldEnforce {
+			mbf.require.True(feeCmp > 0, "expected base fee to be higher than the minBaseFee")
+		} else {
+			mbf.require.True(feeCmp == 0, "expected base fee to be at least minBaseFee")
+		}
+		mbf.t.Logf("base fee %s, minBase %s %d", n.BaseFee(), minBase)
+	}
+}
+
+// waitForMinBaseFee waits until the L2 latest payload extra-data encodes the expected min base fee.
+func (mbf *minBaseFee) waitForMinBaseFee(expected uint64) {
+	client := mbf.l2EL.Escape().L2EthClient()
+	ext, ok := client.(apis.L2EthExtendedClient)
+	mbf.require.True(ok, "L2 client does not support extended payload API")
+
+	expectedExtraData := eth.BytesMax32(eip1559.EncodeJovianExtraData(250, 6, expected))
+
+	var actualPayload eth.BytesMax32
+	mbf.require.Eventually(func() bool {
+		payload, err := ext.PayloadByLabel(mbf.ctx, "latest")
+		if err != nil {
+			return false
+		}
+		if len(payload.ExecutionPayload.ExtraData) != 17 {
+			return false
+		}
+
+		got := binary.BigEndian.Uint64(payload.ExecutionPayload.ExtraData[9:])
+		actualPayload = payload.ExecutionPayload.ExtraData
+		return got == expected
+	}, 2*time.Minute, 5*time.Second, "L2 min base fee did not sync within timeout")
+
+	mbf.require.Equal(expectedExtraData, actualPayload, "extradata doesnt match")
+}
+
+func (mbf *minBaseFee) getBlock() eth.BlockInfo {
+	_ = mbf.l2EL.WaitForBlock()
+	el := mbf.l2EL.Escape().EthClient()
+	info, err := el.InfoByLabel(mbf.ctx, "latest")
+	mbf.require.NoError(err)
+	return info
+}
```

Comment:
I understand the `.WaitForBlock()` helper better now, it does seem to do the right thing. Do we need to wrap it here?